src/app/features/suggested-first-steps/components/add-funds-step.tsx

```diff
@@ -1,24 +1,33 @@
-import { useCallback } from 'react';
+import { useCallback, useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
 
 import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
-import { useSuggestedFirstStepsStatus } from '@app/store/onboarding/onboarding.selectors';
 import AddFundsFull from '@assets/images/onboarding/steps/add-funds-light.png';
 import AddFundsFullDone from '@assets/images/onboarding/steps/add-funds-light-done.png';
 import AddFundsPopup from '@assets/images/onboarding/steps/add-funds-light-sm.png';
 import AddFundsPopupDone from '@assets/images/onboarding/steps/add-funds-light-done-sm.png';
-import { SuggestedFirstSteps, SuggestedFirstStepStatus } from '@shared/models/onboarding-types';
 import { RouteUrls } from '@shared/route-urls';
+import { useCurrentAccountAvailableStxBalance } from '@app/store/accounts/account.hooks';
 
 import { SuggestedFirstStep } from './suggested-first-step';
 
-export function AddFundsStep() {
+const eventName = 'add_funds';
+
+interface AddFundsStepProps {
+  isComplete: boolean;
+  onComplete(): void;
+}
+export function AddFundsStep({ isComplete, onComplete }: AddFundsStepProps) {
   const analytics = useAnalytics();
   const navigate = useNavigate();
-  const stepsStatus = useSuggestedFirstStepsStatus();
+  const availableStxBalance = useCurrentAccountAvailableStxBalance();
+
+  useEffect(() => {
+    if (availableStxBalance?.isGreaterThan(0)) onComplete();
+  }, [availableStxBalance, onComplete]);
```

Comment:
This part confuses me. Why do we ever need to call `onComplete`? This is a state that's kind of updated contingent on the component's lifecycle? Which seems arbitrary.

I'd suggest either:

- The parent component passes in `isComplete={availableStxBalance?.isGreaterThan(0)}`, no `onComplete` callback
- This component accepts balance and handles the logic of whether or not it's "complete"